convert2rhel/unit_tests/actions/conversion/set_efi_config_test.py

```diff
@@ -0,0 +1,240 @@
+    monkeypatch.setattr("os.path.exists", mock.Mock(return_value=False))
+    monkeypatch.setattr("convert2rhel.grub.is_efi", mock.Mock(return_value=True))
+    new_default_efi_bin_instance.run()
+
+    unit_tests.assert_actions_result(
+        new_default_efi_bin_instance,
+        level="ERROR",
+        id="RHEL_UEFI_BINARIES_DO_NOT_EXIST",
+        title="RHEL UEFI binaries do not exist",
+        description="None of the expected RHEL UEFI binaries exist.",
+        diagnosis="The migration of the bootloader setup was not successful.",
+        remediations=(
+            "Do not reboot your machine before doing a manual check of the\n"
+            "bootloader configuration. Ensure that grubenv and grub.cfg files\n"
+            "are present in the %s directory and that\n"
+            "a new bootloader entry for Red Hat Enterprise Linux exists\n"
+            "(check `efibootmgr -v` output).\n"
+            "The entry should point to '\\EFI\\redhat\\shimx64.efi'." % grub.RHEL_EFIDIR_CANONICAL_PATH
+        ),
+    )
+
+
+def test_efi_bootmgr_utility_installed_error(efi_bootmgr_utility_installed_instance, monkeypatch):
+    monkeypatch.setattr("os.path.exists", mock.Mock(return_value=False))
+    efi_bootmgr_utility_installed_instance.run()
+
+    unit_tests.assert_actions_result(
+        efi_bootmgr_utility_installed_instance,
+        level="ERROR",
+        id="EFIBOOTMGR_UTILITY_NOT_INSTALLED",
+        title="Efibootmgr utility is not installed",
+        description="The /usr/sbin/efibootmgr utility is not installed.",
+        remediations="Install the efibootmgr utility via YUM/DNF.",
+    )
+
+
+@pytest.mark.parametrize(
+    ("sys_id", "src_file_exists", "dst_file_exists", "log_msg"),
+    (
+        ("oracle", None, None, "only related to CentOS Linux"),
+        ("centos", None, True, "file already exists"),
+        ("centos", True, False, "Copying '"),
+    ),
+)
+@mock.patch("shutil.copy2")
+@mock.patch("os.path.exists")
+def test__copy_grub_files(
+    mock_path_exists,
+    mock_path_copy2,
+    sys_id,
+    src_file_exists,
+    dst_file_exists,
+    log_msg,
+    monkeypatch,
+    caplog,
+    copy_grub_files_instance,
+):
+    def path_exists(path):
+        return src_file_exists if grub.CENTOS_EFIDIR_CANONICAL_PATH in path else dst_file_exists
+
+    mock_path_exists.side_effect = path_exists
+    monkeypatch.setattr("convert2rhel.systeminfo.system_info.id", sys_id)
+
+    copy_grub_files_instance.run()
+    assert any(log_msg in record.message for record in caplog.records)
+    if sys_id == "centos" and src_file_exists and not dst_file_exists:
+        assert mock_path_copy2.call_args_list == [
+            mock.call("/boot/efi/EFI/centos/grubenv", "/boot/efi/EFI/redhat/grubenv"),
+            mock.call("/boot/efi/EFI/centos/grub.cfg", "/boot/efi/EFI/redhat/grub.cfg"),
+            mock.call("/boot/efi/EFI/centos/user.cfg", "/boot/efi/EFI/redhat/user.cfg"),
+        ]
```

Comment:
```suggestion
@pytest.mark.parametrize(
    ("sys_id", "src_file_exists", "dst_file_exists", "log_msg"),
    (
        ("oracle", None, None, "only related to CentOS Linux"),
        ("centos", None, True, "file already exists"),
        ("centos", True, False, "Copying '"),
    ),
)
def test__copy_grub_files(
    sys_id,
    src_file_exists,
    dst_file_exists,
    log_msg,
    monkeypatch,
    caplog,
    copy_grub_files_instance,
    global_system_info,
):
    def path_exists(path):
        return src_file_exists if grub.CENTOS_EFIDIR_CANONICAL_PATH in path else dst_file_exists
    monkeypatch.setattr(os.path, "exists", path_exists)
    monkeypatch.setattr(shutil, "copy2", mock.Mock())
    global_system_info.id = sys_id

    copy_grub_files_instance.run()
    assert any(log_msg in record.message for record in caplog.records)
    if sys_id == "centos" and src_file_exists and not dst_file_exists:
        assert mock_path_copy2.call_args_list == [
            mock.call("/boot/efi/EFI/centos/grubenv", "/boot/efi/EFI/redhat/grubenv"),
            mock.call("/boot/efi/EFI/centos/grub.cfg", "/boot/efi/EFI/redhat/grub.cfg"),
            mock.call("/boot/efi/EFI/centos/user.cfg", "/boot/efi/EFI/redhat/user.cfg"),
        ]
```

We might need to check on this one if global_system_info might not need to be added as a monkeypatch in this case. I think it will probably not need.